packages/kit/types/index.d.ts

```diff
@@ -865,7 +865,7 @@ export interface RequestEvent<
 	/**
 	 * Additional data made available through the adapter.
 	 */
-	platform: Readonly<App.Platform>;
+	platform: Readonly<App.Platform> | void;
```

Comment:
I think this will be super annoying for type safety, AFAIK you can't use the `?` operator with it. `void` means something like "do not use this return value, could be anything"

Reply:
Not sure I follow? [This](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgAoBs5hgeygW2QG8AoZc5fCAZ2rgHMIAuZasKUegbhIF8SSoSLEQoAShACOAVxpgAogDcI4YmQoAHTNjz4WEuABMcIdAE8APBiy4CAPmQAfZIpzBDPfiRjSQCMMAmyIwKyuAAFACU+lKybEoqYGoUyFAQYNJQIMkpmtq2esi+hhAwoBCG6uS8ngIIJmzIKAC8wekJEZE8JPUg1DjoEAB06Dj04cNaNroA-ENUtAwQXUA) seems to work fine:

```ts
interface Platform {
    message: string;
}

interface RequestEvent {
    platform: Readonly<Platform> | void;
}

function getEvent(): RequestEvent {
    return {
        platform: undefined
    };
}

const e = getEvent();

console.log(e.platform?.message);
```

Reply:
Mhm might have mixed up something, nevermind then (though I still feel `undefined` is the better type)

Reply:
Is there a reason it's better? I feel confident that I've run into cases before where `undefined` caused problems that were solved by using `void` instead, though I can't recall the specifics

Reply:
My understanding is that `void` actually means "won't be used used", so in the context of a function return it implies that nothing will be explicitly returned but doesn't enforce that the returned value will be `undefined`, which means that the following code is okay with TS.

```ts
const v: () => void = () => 1
``` 

In this context here, a property assignment, I believe that `void` and `undefined` have the same outcome, but that `undefined` is semantically more typical.

Reply:
fair enough, I'll change it back

```suggestion
	platform: Readonly<App.Platform> | undefined;
```